File: WinForms/CS/TreeListMainDemo/Options/ucScrollAnnotationsOptions.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using DevExpress.Utils;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Annotations;

namespace DevExpress.XtraTreeList.Demos.Options {
    public partial class ucScrollAnnotationsOptions : ucDefault {
        public ucScrollAnnotationsOptions() {
            InitializeComponent();
            stylesSource.DataSource = style;
        }
        readonly AnnotationsStyle style = new AnnotationsStyle();
        public bool BookmarksEnabled {
            get { return ceBookmarksEnabled.Checked; }
        }
        public Color GetColor(ScrollAnnotationKind kind) {
            return style.GetColor(kind);
        }
        //<ucScrollAnnotationsOptions>
        void OnAnnotationsEnabledChanged(object sender, EventArgs e) {
            bool _enabled = ((CheckEdit)sender).Checked;
            DefaultBoolean _show = _enabled ? DefaultBoolean.True : DefaultBoolean.False;
            if(TreeList == null) return;
            if(sender == ceErrorsEnabled) {
                layoutControlItemForErrors.Enabled = _enabled;
                TreeList.OptionsScrollAnnotations.ShowErrors = _show;
            }
            if(sender == ceFocusedEnabled) {
                layoutControlItemForFocused.Enabled = _enabled;
                TreeList.OptionsScrollAnnotations.ShowFocusedRow = _show;
            }
            if(sender == ceBookmarksEnabled)
                TreeList.OptionsScrollAnnotations.ShowCustomAnnotations = _show;
        }
        void OnAnnotationsStyleChanged(object sender, ListChangedEventArgs e) {
            if(e.ListChangedType == ListChangedType.ItemChanged)
                TreeList.InvalidateScrollAnnotations();
        }
        //</ucScrollAnnotationsOptions>
    }
    public class AnnotationsStyle {
        readonly IDictionary<ScrollAnnotationKind, Color> colors = new Dictionary<ScrollAnnotationKind, Color>();
        public Color ErrorsColor {
            get { return GetColor(ScrollAnnotationKind.Errors); }
            set { colors[ScrollAnnotationKind.Errors] = value; }
        }
        public Color SearchResultsColor {
            get { return GetColor(ScrollAnnotationKind.SearchResults); }
            set { colors[ScrollAnnotationKind.SearchResults] = value; }
        }
        public Color FocusColor {
            get { return GetColor(ScrollAnnotationKind.FocusedRow); }
            set { colors[ScrollAnnotationKind.FocusedRow] = value; }
        }
        public Color GetColor(ScrollAnnotationKind kind) {
            Color color;
            return colors.TryGetValue(kind, out color) ? color : Color.Empty;
        }
    }
}

File: WinForms/VB/TreeListMainDemo/Options/ucScrollAnnotationsOptions.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports DevExpress.Utils
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Annotations

Namespace DevExpress.XtraTreeList.Demos.Options

    Public Partial Class ucScrollAnnotationsOptions
        Inherits ucDefault

        Public Sub New()
            InitializeComponent()
            stylesSource.DataSource = style
        End Sub

        Private ReadOnly style As AnnotationsStyle = New AnnotationsStyle()

        Public ReadOnly Property BookmarksEnabled As Boolean
            Get
                Return ceBookmarksEnabled.Checked
            End Get
        End Property

        Public Function GetColor(ByVal kind As ScrollAnnotationKind) As Color
            Return style.GetColor(kind)
        End Function

        '<ucScrollAnnotationsOptions>
        Private Sub OnAnnotationsEnabledChanged(ByVal sender As Object, ByVal e As EventArgs)
            Dim _enabled As Boolean = CType(sender, CheckEdit).Checked
            Dim _show As DefaultBoolean = If(_enabled, DefaultBoolean.True, DefaultBoolean.False)
            If TreeList Is Nothing Then Return
            If sender Is ceErrorsEnabled Then
                layoutControlItemForErrors.Enabled = _enabled
                TreeList.OptionsScrollAnnotations.ShowErrors = _show
            End If

            If sender Is ceFocusedEnabled Then
                layoutControlItemForFocused.Enabled = _enabled
                TreeList.OptionsScrollAnnotations.ShowFocusedRow = _show
            End If

            If sender Is ceBookmarksEnabled Then TreeList.OptionsScrollAnnotations.ShowCustomAnnotations = _show
        End Sub

        Private Sub OnAnnotationsStyleChanged(ByVal sender As Object, ByVal e As ListChangedEventArgs)
            If e.ListChangedType = ListChangedType.ItemChanged Then TreeList.InvalidateScrollAnnotations()
        End Sub
    '</ucScrollAnnotationsOptions>
    End Class

    Public Class AnnotationsStyle

        Private ReadOnly colors As IDictionary(Of ScrollAnnotationKind, Color) = New Dictionary(Of ScrollAnnotationKind, Color)()

        Public Property ErrorsColor As Color
            Get
                Return GetColor(ScrollAnnotationKind.Errors)
            End Get

            Set(ByVal value As Color)
                colors(ScrollAnnotationKind.Errors) = value
            End Set
        End Property

        Public Property SearchResultsColor As Color
            Get
                Return GetColor(ScrollAnnotationKind.SearchResults)
            End Get

            Set(ByVal value As Color)
                colors(ScrollAnnotationKind.SearchResults) = value
            End Set
        End Property

        Public Property FocusColor As Color
            Get
                Return GetColor(ScrollAnnotationKind.FocusedRow)
            End Get

            Set(ByVal value As Color)
                colors(ScrollAnnotationKind.FocusedRow) = value
            End Set
        End Property

        Public Function GetColor(ByVal kind As ScrollAnnotationKind) As Color
            Dim color As Color
            Return If(colors.TryGetValue(kind, color), color, Color.Empty)
        End Function
    End Class
End Namespace
